server/app/controllers/admin/AdminApplicationController.java

```diff
@@ -148,16 +154,41 @@ public Result downloadDemographics() {
         .withHeader("Content-Disposition", String.format("attachment; filename=\"%s\"", filename));
   }
 
-  /** Download a PDF file of the application to the program. This feature is not implemented yet. */
+  /** Download a PDF file of the application to the program. */
   @Secure(authorizers = Authorizers.Labels.ANY_ADMIN)
   public Result download(Http.Request request, long programId, long applicationId) {
     try {
       ProgramDefinition program = programService.getProgramDefinition(programId);
       checkProgramAdminAuthorization(profileUtils, request, program.adminName()).join();
-      throw new UnsupportedOperationException("Not yet implemented.");
     } catch (ProgramNotFoundException e) {
       return notFound(e.toString());
+    } catch (CompletionException e) {
+      return unauthorized();
+    }
+
+    Optional<Application> applicationMaybe =
+        this.applicationRepository.getApplication(applicationId).toCompletableFuture().join();
+
+    if (!applicationMaybe.isPresent()) {
+      return notFound(String.format("Application %d does not exist.", applicationId));
+    }
+
+    Application application = applicationMaybe.get();
+    String applicantNameWithApplicationId =
+        String.format("%s (%d)", application.getApplicantData().getApplicantName(), application.id);
+
+    String filename =
+        String.format("%s-%s.pdf", applicantNameWithApplicationId, clock.instant().toString());
+    byte[] pdf = null;
+    try {
+      pdf = pdfExporter.export(application, applicantNameWithApplicationId);
+    } catch (DocumentException | IOException e) {
+      throw new RuntimeException(
+          "Document or IOException exception detected while processing the PDF export!");
```

Comment:
suggestion (blocking): include the original exception as the cause
```suggestion
      throw new RuntimeException(e);
```